frontend/src/metabase/containers/ItemPicker/Item.styled.tsx

```diff
@@ -53,6 +57,7 @@ export const ExpandButton = styled(IconButtonWrapper)<{ canSelect: boolean }>`
   border: 1px solid ${color("border")};
 
   &:hover {
+    color: ${color("brand")} !important;
```

Comment:
![image](https://github.com/metabase/metabase/assets/6830683/85bc131c-3aef-4346-9681-d4a25e3604ce)

1. :x: The issue is still there when `props.canSelect` is `false` and the "ExpandButton" is hovered (see screenshot). We need to take `props.canSelect` into account, just like in case of `background-color` a line below.

2. Can we avoid using `!important`?

Reply:
Hi @kamilmielnik , I've updated the styles based on your feedback as seen below:

|                   | Before | After |
|-------------------|--------|-------|
| `canSelect=false` |  ![before_cant-select_hover](https://github.com/metabase/metabase/assets/15979404/c308ab8c-6e99-45ca-b371-bea5eee7b326)  |   ![after_cant-select_hover](https://github.com/metabase/metabase/assets/15979404/41a491fe-3282-4f9c-81a9-07f36fb916d4) |
| `selected=true`   |  ![before_selected](https://github.com/metabase/metabase/assets/15979404/6ef948f0-0865-4e88-bcbd-6797d449fdf1) |  ![after_selected](https://github.com/metabase/metabase/assets/15979404/2e7c5cf1-448e-43ac-8428-53def75c2173) |

I've also removed the `!important` property and use `ExpandButton > &:hover > & svg.Icon` instead to change the color of the icon from the `ExpandButton`'s styles.

Thanks for your feedback!

Reply:
Clicking the right-arrow-button always navigates into the collection. When right-arrow-button is hovered, its background is white and foreground is blueish.

When `canSelect` is `true`, clicking the collection item selects the collection.
But when `canSelect` is `false`, clicking the collection item will navigate into that collection, just as if the right-arrow-button was clicked. For that reason, in that case (i.e. `canSelect` is `false` and collection item is hovered), I believe the right-arrow-button should be styled with white background and blueish foreground, to indicate that the navigation is about to happen (consistently with hovering the right-arrow-button).

I hope that makes sense.

Reply:
Looks great!
Good call on adding the explanatory comment :+1: